### ISSUE TITLE
Querying transaction hash via REST server

### ISSUE THREAD

**Issue by maddyonline** [cosmos/sdk-tutorials#131]
Am I correct to believe that [rest server](https://github.com/cosmos/sdk-application-tutorial/blob/master/tutorial/run-rest.md) that is started using `nscli` is meant to roughly mimic  [gaicli rest server](https://cosmos.network/docs/clients/service-providers.html#setting-up-the-rest-server)? In particular, should it fully support at least ICS0 Tendermint APIs on this [Swagger page](https://cosmos.network/rpc/)?

The API that is missing is `/txs/{hash}` (Get a Tx by hash). I am able to use this API via `gaicli` rest server but not via `nscli` rest server.

**Comment by tuckermint** [comment]
I see tx routes were added like this: https://github.com/cosmos/sdk-tutorials/commit/0b2687e6a6bcc8ce421c53b83de1478fd8a56729

I need RPC Routes added from here:
https://github.com/cosmos/cosmos-sdk/blob/master/client/rpc/root.go

Can I just do the same thing with RegisterRPCRoutes as you did with RegisterTxRoutes?

Or do I also need to register the rpc module in app.go like the auth module?

Thanks!
@marbar3778

**Comment by marbar3778** [comment]
hey sorry for the delay on solving this issue. I am waiting for 0.38 to be released then I will go through and update everything and cleanup things in one swoop. the release is aimed for the coming days.

**Comment by tuckermint** [comment]
v0.38.0 is live:
https://github.com/cosmos/cosmos-sdk/releases

**Comment by marbar3778** [comment]
yes as you can see here: #241 I have been working on updating the tutorials

**Comment by tuckermint** [comment]
Is the current code in this branch safe to use for a live chain:
https://github.com/cosmos/sdk-tutorials/tree/marko/update_tutorials

Or should I wait for it to be merged into master?

**Comment by marbar3778** [comment]
it is safe. I am waiting on the hellochain markdown to be updated in order to merge the code

**Comment by marbar3778** [comment]
If you are running a live chain I would recommend another couple module in order to bring the secruity of the chain to a higher level
